File: clothing_app_backend/src/bannerModule/banner.model.ts
import mongoose from "mongoose";

const BANNER_TYPES = [
  'featured', 'popular_category', 'chic_look', 'seasonal', 'discount',
  'new_arrival', 'best_seller', 'clearance', 'you_may_like', 'model_list',
  'all_products', 'flash_sale', 'editor_pick', 'trending_now', 'limited_edition',
  'back_in_stock', 'shop_by_occasion'
];

const bannerSchema = new mongoose.Schema({
  title: { type: String, required: true },
  description: { type: String, default: '' },
  image: { type: String, required: true }, // base64 string
  type: { type: String, enum: BANNER_TYPES, required: true },
  redirect_url: { type: String, default: null },
  is_active: { type: Boolean, default: true },
  priority: { type: Number, default: 0 },
  created_at: { type: Date, default: Date.now },
  updated_at: { type: Date, default: Date.now }
});

const Banner = mongoose.model(
  "Banner",
  bannerSchema
);
export default Banner;

File: clothing_app_frontend/lib/homeModule/model/banner_model.dart
class Banner {
  String? id;
  String? title;
  String? description;
  String? image;
  String? type;
  bool? isActive;
  DateTime? createdAt;
  DateTime? updatedAt;

  Banner({
    this.id,
    this.title,
    this.description,
    this.image,
    this.type,
    this.isActive,
    this.createdAt,
    this.updatedAt,
  });

  static Banner jsonToBanner(Map<String, dynamic> json) {
    return Banner(
      id: json['_id']?.toString(),
      title: json['title']?.toString(),
      description: json['description']?.toString(),
      image: json['image']?.toString(),
      type: json['type']?.toString(),
      isActive: json['is_active'] as bool?,
      createdAt: json['createdAt'] != null
          ? DateTime.tryParse(json['createdAt'].toString())
          : null,
      updatedAt: json['updatedAt'] != null
          ? DateTime.tryParse(json['updatedAt'].toString())
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'title': title,
      'description': description,
      'image': image,
      'type': type,
      'is_active': isActive,
      'createdAt': createdAt?.toIso8601String(),
      'updatedAt': updatedAt?.toIso8601String(),
    };
  }

  @override
  String toString() {
    return 'Banner{id: $id, title: $title, type: $type, isActive: $isActive}';
  }
}
